scripts/main.ts

```diff
@@ -37,34 +37,200 @@ const AutocratIDL: AutocratV0 = require("../target/idl/autocrat_v0.json");
+    icon: 'deanToken.png',
+    devnetBaseToken: DEAN_DEVNET,
+    mainnetBaseToken: DEAN,
+    tokenSymbol: 'DEAN',
+    publicKey: new PublicKey("5rhzhBtevKSjuFMTL377M1AWRsiTF57p9J88TzptzUPp"),
+  },
+];
+
+const CONDITIONAL_VAULT_VERSIONS = [
+  {
+    version: 0,
+    programId: new PublicKey("vaU1tVLj8RFk7mNj1BxqgAsMKKaL8UvEUHvU3tdbZPe")
+  },{
+    version: 2,
+    programId: new PublicKey("vAuLTQjV5AZx5f3UgE75wcnkxnQowWxThn1hGjfCVwP")
+  },
+];
+
+const OPENBOOK_TWAP_VERSIONS = [
+  {
+    version: 0,
+    programId: new PublicKey("TWAP7frdvD3ia7TWc8e9SxZMmrpd2Yf3ifSPAHS8VG3")
+  },{
+    version: 1,
+    programId: new PublicKey("TWAPrdhADy2aTKN5iFZtNnkQYXERD9NvKjPFVPMSCNN")
+  },{
+    version: 2,
+    programId: new PublicKey("twAP5sArq2vDS1mZCT7f4qRLwzTfHvf5Ay5R5Q5df1m")
+  },
+]
+
+const AUTOCRAT_MIGRATOR_VERSIONS = [
+  {
+    version: 1,
+    programId: new PublicKey("migkwAXrXFN34voCYQUhFQBXZJjHrWnpEXbSGTqZdB3")
+  },{
+    version: 2,
+    programId: new PublicKey("MigRDW6uxyNMDBD8fX2njCRyJC4YZk2Rx9pDUZiAESt")
+  },
+]
+
+// Used with setting for version
+const useVersion = 3 // Which version you want to use scripts with
+const useDao = 'deanslist'
+const useNet = 'devnet'
+
+// Setup for returning from our selection above
+const AUTOCRAT = AUTOCRAT_VERSIONS.find((program) => program.version === useVersion && (program.version > 2 || program.dao === useDao))
+// Filter through our DAOs for MultiDAO
+export const DAO = DAOS.find((dao) => dao.daoName.toLowerCase() === useDao.toLowerCase())
+// Setup for fetching baseToken given the version and network
+let baseToken: PublicKey;
+if (useNet === 'devnet') {
+  baseToken = useVersion < 3 ? AUTOCRAT.devnetBaseToken : DAO.devnetBaseToken
+} else {
+  baseToken = useVersion < 3 ? AUTOCRAT.mainnetBaseToken: DAO.mainnetBaseToken
+}
+export const BASE_TOKEN = baseToken;
+export const QUOTE_TOKEN = useNet === 'devnet' ? AUTOCRAT.devnetQuoteToken : AUTOCRAT.mainnetQuoteToken
+// These programs don't always have upgrades, therefore you want to
+// use the matching version or one version less.
+// eg. v0.3 autocrat uses v0.2 twap version, or v0.1 uses v0 conditional vault
+// NOTE: The only exception is the migrator, which extends one beyond
+// eg. v0 autocrat uses v0.1 autocrat migrator
+// TODO: We need some logic to loop through and find highest version number and then step back one
+const CONDITIONAL_VAUT = CONDITIONAL_VAULT_VERSIONS.find(
+  (program) => 
+    // NOTE: This is a quick and dirty way
+    program.version === useVersion || program.version === useVersion - 1 || program.version
+)
+const OPENBOOK_TWAP = OPENBOOK_TWAP_VERSIONS.find(
+  (program) => 
+    // NOTE: This is a quick and dirty way
+    program.version === useVersion || program.version === useVersion - 1 || program.version
+)
+const AUTOCRAT_MIGRATOR = AUTOCRAT_MIGRATOR_VERSIONS.find(
+  // TODO: Hack to get this to play nice...
+  (program) => program.version === useVersion + 1 || program.version === useVersion - 1
```

Comment:
This has a bug, in that it is looping from index 0, and really we want the greatest if we can find it, not the first one it finds.